Sources/Socket.swift

```diff
@@ -104,6 +104,37 @@ class Socket {
     }
   }
 
+  func bind(path: String) throws {
+    var addr = sockaddr_un()
+    addr.sun_family = sa_family_t(AF_UNIX)
+
+    // addr.sun_path is a `char [N]` and become a tuple in Swift.
+    // Its size is different on different platforms.
+    // Details can be found in `man 7 unix`
```

Comment:
I think we should instead dynamically figure out the size at compile time rather than hard-code it into the source.

The following should work on Linux, we would need to found in `sun_len` on OS X at least.

```swift
sizeof(sockaddr_un) - sizeof(sa_familt_t)
```